definitions/marts/looker_studio_marts/npq/ls_npq_retention_kpi_cube.sqlx

```diff
@@ -0,0 +1,87 @@
+config {
+    type: "table",
+    bigquery: {
+        partitionBy: "",
+        clusterBy: ["provider_name", "cohort_number", "short_course_name"]
+    },
+    description: "This mart is designed as a cube to easily calculate retention KPIs for DfE funded NPQ applicants. One row represents a lead provier name, cohort number and short course name combination with the metrics: total_application_count, completed_count, active_count, deferred_count, withdrawn_count and retention_kpi_percentage calcualted for each.",
+    columns: {
+        provider_name: "Name of the provider.",
+        cohort_number: "A policy orientated cohort identifier to indicate which cohort a DfE funded applicants lies, this field is calcualated through a combination of cohort years and in the years 2021-2023 which 'schedule_identifier' they will be a part of.",
+        short_course_name: "Shortened name of the course.",
+        completed_count: "The count of DfE funded applicants which have completed their NPQ course. Completed NPQ courses are defined as where the retention_status is 'eligible', 'payable' or 'paid'.",
+        active_count: "The count of DfE funded applicants which are undertaking their NPQ course.",
+        deferred_count: "The count of DfE funded applicants which have deferred their NPQ course.",
+        withdrawn_count: "The count of DfE funded applicants which have withdrawn from their NPQ course.",
+        total_application_count: "The total number of DfE funded applications recieved",
+        retention_kpi_percentage: "Retention is defined as the retention rate for all completed and active counts, deferred statuses are excluded from this metric. retention_rate = (completed_count+active_count)/(completed_count+active_count+withdrawn)."

```

Comment:
@t-phillips-dfe if you want this to be a cube then only include metrics in it that you can aggregate usefully (e.g. sum or average) across the different rows. retention_kpi_percentage looks to me like you can't aggregate it usefully - if you add it up across rows you'll get something that's obviously wrong; if you take an average of its values across a set of rows then you'll get something that looks misleadingly like it might be right but in fact isn't.

To include a % in a cube like this, instead just include the numerator and denominator as separate fields. Then define a metric in the Looker Studio data source which SUMs them each up separately and then divides.